src/pages/Task.tsx

```diff
@@ -185,6 +164,15 @@ const TaskCore: React.FC = () => {
         banners={bannersState}
         removeBanner={dispatchBanner.removeBanner}
       />
+      {failedTask && execution !== undefined ? (
+        <BuildBaronVariables
```

Comment:
This function component can be converted into a react hook that returns `{showBuildBaronTab, buildBaronData, buildBaronError}`. In order to do that, you would have to handle this case `failedTask && execution !== undefined` within the hook because hooks cannot be conditionally rendered. 
A technicality that will come up when trying to do this is that we do not want to make buildBaron queries for failed tasks or when the execution is undefined. We can circumvent that by importing `useLazyQuery` instead of `useQuery` and only fetching/refetching when the props failedTask and execution are valid (true, defined).